File: 1284-four-divisors/four-divisors.py
class Solution(object):
    def sumFourDivisors(self, nums):
        res=0
       
        for i in nums:
            l=set()
            for j in range(1,int(math.sqrt(i))+1):
                if(i%j==0):
                    l.add(j)
                    l.add(i//j)
            if(len(l)==4):
                for k in l:
                    res+=k
        return res

        
        

File: 1284-four-divisors/four-divisors.java
class Solution {
    public int sumFourDivisors(int[] nums) {
        int sum=0;
        for(int i:nums)
        {
            Set<Integer> l=new HashSet <Integer>();
            for(int j=1;j<=Math.sqrt(i);j++)
            {
                if(i%j==0){
                l.add(j);
                l.add((int)i/j);
                }
            }
            if(l.size()==4)
            for(int k:l)
            sum+=k;
        }
        return sum;
    }
}